addon/utils/notify-changes.js

```diff
@@ -0,0 +1,53 @@
+import Ember from 'ember';
+import { notifyPropertyChange as _notifyPropertyChange } from '@ember/object';
+import { assert } from '@ember/debug';
+import { DEBUG } from '@glimmer/env';
+
+const { changeProperties } = Ember;
+
+const { propertyDidChange } = Ember;
+export let notifyPropertyChange;
+
+const HasNotifyPropertyChange = _notifyPropertyChange !== undefined;
+if (HasNotifyPropertyChange) {
+  notifyPropertyChange = _notifyPropertyChange;
+} else {
+  notifyPropertyChange = propertyDidChange;
+}
+
+const StoreToChanges = new WeakMap();
+
+export function beginChanges(store) {
+  StoreToChanges.set(store, []);
```

Comment:
Whats the advantage of this approach vs keeping track of this on the store?

Reply:
not exposing private api to users